bridges/ethMultiversX/bridgeExecutor_test.go

```diff
@@ -1006,12 +1009,151 @@ func TestMultiversXToEthBridgeExecutor_GetAndStoreBatchFromMultiversX(t *testing
+		assert.Nil(t, err)
+
+		err = executor.StoreBatchFromMultiversX(batch)
+		assert.Equal(t, providedBatchWithDeposits, executor.batch)
+		assert.Nil(t, err)
+	})
+	t.Run("should add deposits metadata for sc calls", func(t *testing.T) {
+		t.Parallel()
+
+		providedNonce := uint64(8346)
+		depositNonce := uint64(100)
+		depositData := []byte("testData")
+		expectedBatch := &bridgeCore.TransferBatch{
+			ID: providedNonce,
+			Deposits: []*bridgeCore.DepositTransfer{
+				{
+					Nonce: depositNonce,
+				},
+			},
+		}
+		args := createMockExecutorArgs()
+		args.MultiversXClient = &bridgeTests.MultiversXClientStub{
+			GetPendingBatchCalled: func(ctx context.Context) (*bridgeCore.TransferBatch, error) {
+				return expectedBatch, nil
+			},
+			GetBatchSCMetadataCalled: func(ctx context.Context, batch *bridgeCore.TransferBatch) ([]*transaction.Events, error) {
+				depositNonceBytes := make([]byte, 8)
+				binary.BigEndian.PutUint64(depositNonceBytes, depositNonce)
+				return []*transaction.Events{
+					{Topics: [][]byte{{}, depositNonceBytes, {}, {}, {}, {}, {}, depositData}},
+				}, nil
+			},
+		}
+
+		executor, _ := NewBridgeExecutor(args)
+		batch, err := executor.GetBatchFromMultiversX(context.Background())
+		assert.Nil(t, err)
+		assert.Equal(t, expectedBatch, batch)
+		expectedDepositData := []byte{bridgeCore.DataPresentProtocolMarker, 0, 0, 0, byte(len(depositData))}
+		expectedDepositData = append(expectedDepositData, depositData...)
+		assert.Equal(t, string(expectedDepositData), string(batch.Deposits[0].Data))
+	})
+	t.Run("should add deposits metadata for sc calls with a data starting with missing data marker", func(t *testing.T) {
+		t.Parallel()
+
+		providedNonce := uint64(8346)
+		depositNonce := uint64(100)
+		depositData := append([]byte{bridgeCore.MissingDataProtocolMarker}, "testData"...)
+		expectedBatch := &bridgeCore.TransferBatch{
+			ID: providedNonce,
+			Deposits: []*bridgeCore.DepositTransfer{
+				{
+					Nonce: depositNonce,
+				},
+			},
+		}
+		args := createMockExecutorArgs()
+		args.MultiversXClient = &bridgeTests.MultiversXClientStub{
+			GetPendingBatchCalled: func(ctx context.Context) (*bridgeCore.TransferBatch, error) {
+				return expectedBatch, nil
+			},
+			GetBatchSCMetadataCalled: func(ctx context.Context, batch *bridgeCore.TransferBatch) ([]*transaction.Events, error) {
+				depositNonceBytes := make([]byte, 8)
+				binary.BigEndian.PutUint64(depositNonceBytes, depositNonce)
+				return []*transaction.Events{
+					{Topics: [][]byte{{}, depositNonceBytes, {}, {}, {}, {}, {}, depositData}},
+				}, nil
+			},
+		}
+
+		executor, _ := NewBridgeExecutor(args)
+		batch, err := executor.GetBatchFromMultiversX(context.Background())
 		assert.Nil(t, err)
 
 		err = executor.StoreBatchFromMultiversX(batch)
-		assert.Equal(t, providedBatch, executor.batch)
 		assert.Nil(t, err)
+		assert.True(t, expectedBatch == executor.GetStoredBatch()) // pointer testing
```

Comment:
irrelevant for the test. Why not construct the expectedBatch to hold also the expected deposits? The test will be clearer